From 15ff5608d61ffdf3e8e6bdfd4e16d7d6ea721b2d Mon Sep 17 00:00:00 2001
From: Khanh Le <khanh.le.xr@renesas.com>
Date: Tue, 12 Aug 2025 18:14:09 +0700
Subject: [PATCH] clk: renesas: r8a779f0: Add WWDT and parent SASYNC clocks

Adapted from Linux commit ca5f7743eb62
("clk: renesas: r8a779f0: Add WWDT and parent SASYNC clocks")

Add WWDT and parent SASYNC clocks

Signed-off-by: Khanh Le <khanh.le.xr@renesas.com>
---
 drivers/clk/renesas/r8a779f0-cpg-mssr.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/clk/renesas/r8a779f0-cpg-mssr.c b/drivers/clk/renesas/r8a779f0-cpg-mssr.c
index d4a178fb208..3dbfff32154 100644
--- a/drivers/clk/renesas/r8a779f0-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a779f0-cpg-mssr.c
@@ -123,6 +123,7 @@ static const struct cpg_core_clk r8a779f0_core_clks[] = {
 	DEF_FIXED("rsw2",	R8A779F0_CLK_RSW2,	CLK_PLL5_DIV2,	5, 1),
 	DEF_FIXED("cbfusa",	R8A779F0_CLK_CBFUSA,	CLK_EXTAL,	2, 1),
 	DEF_FIXED("cpex",	R8A779F0_CLK_CPEX,	CLK_EXTAL,	2, 1),
+	DEF_FIXED("sasyncrt",   R8A779F0_CLK_SASYNCRT,  CLK_PLL5_DIV4,  48, 1),
 	DEF_FIXED("sasyncperd1", R8A779F0_CLK_SASYNCPERD1, CLK_SASYNCPER, 1, 1),
 	DEF_FIXED("sasyncperd2", R8A779F0_CLK_SASYNCPERD2, CLK_SASYNCPER, 2, 1),
 	DEF_FIXED("sasyncperd4", R8A779F0_CLK_SASYNCPERD4, CLK_SASYNCPER, 4, 1),
@@ -175,6 +176,16 @@ static const struct mssr_mod_clk r8a779f0_mod_clks[] = {
 	DEF_MOD("cmt2",		912,	R8A779F0_CLK_R),
 	DEF_MOD("cmt3",		913,	R8A779F0_CLK_R),
 	DEF_MOD("pfc0",		915,	R8A779F0_CLK_CL16M),
+	DEF_MOD("wwdt0",	1318,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt1",	1319,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt2",	1320,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt3",	1321,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt4",	1322,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt5",	1323,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt6",	1324,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt7",	1325,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt8",	1326,	R8A779F0_CLK_SASYNCRT),
+	DEF_MOD("wwdt9",	1327,	R8A779F0_CLK_SASYNCRT),
 	DEF_MOD("rsw2",		1505,	R8A779F0_CLK_RSW2),
 	DEF_MOD("ethphy",	1506,	R8A779F0_CLK_S0D2),
 	DEF_MOD("ufs0",		1514,	R8A779F0_CLK_S0D4),
-- 
2.34.1

